### DISCUSSION TITLE
Dall-E Plugin not working consistently

### DISCUSSION THREAD

**Discussion by points-unknown** [danny-avila/LibreChat#1406] [Q&A] [answered]
I am sure I have something not set up correctly, but I can't get any of the plugins to work for me. I appear to have some things working, I can get GPT3.5 responses. and GPT 4 vision looks to be working, but when I try to use the Dall-E  plugin (or Dall-E-3) I get errors.

Whatever my issue is, it seems to be hit and miss. I did a very simple prompt and got an image:

![image](https://github.com/danny-avila/LibreChat/assets/34283742/1d503566-a3fa-4477-b511-8d0350307fb6)

Then a more complex prompt. and got this:
![image](https://github.com/danny-avila/LibreChat/assets/34283742/7f0f6db3-0942-44dd-bb7a-a49729e686ec)

I am running LibreChat in docker . This is what I see in the log:
```

"lc": 1,
LibreChat      |         "type": "constructor",
LibreChat      |         "id": [
LibreChat      |           "langchain",
LibreChat      |           "schema",
LibreChat      |           "AIMessage"
LibreChat      |         ],
LibreChat      |         "kwargs": {
LibreChat      |           "content": "",
LibreChat      |           "additional_kwargs": {
LibreChat      |             "function_call": {
LibreChat      |               "name": "dalle",
LibreChat      |               "arguments": "{\n  \"prompt\": \"A super realistic 4k cinematic image of an alien ship with thrusters and lights landing on Mount Adams in the state of Washington, amidst the tall fir trees.\",\n  \"style\": \"vivid\",\n  \"quality\": \"hd\",\n  \"size\": \"3840x2160\"\n}"
LibreChat      |             }
LibreChat      |           }
LibreChat      |         }
LibreChat      |       },
LibreChat      |       {
LibreChat      |         "lc": 1,
LibreChat      |         "type": "constructor",
LibreChat      |         "id": [
LibreChat      |           "langchain",
LibreChat      |           "schema",
LibreChat      |           "FunctionMessage"
LibreChat      |         ],
LibreChat      |         "kwargs": {
LibreChat      |           "content": "Please try again, use an API function call with the correct properties/parameters",
LibreChat      |           "name": "dalle",
LibreChat      |           "additional_kwargs": {}
LibreChat      |         }
LibreChat      |       },
LibreChat      |       {
```

I see when folks have issues with Dall-E there is talk of using a reverse proxy? but I'll admit I dont know what that means. I am connecting to LibreChat on my server via Traefik and cloudflare. But I have tried a local connection via my IP and have the same issue. I do not have any proxies entered in the .env.

Really appreciate any help or insight someone can offer.

Relevant .env excerpts....:
```
#==================================================#
#               Server Configuration               #
#==================================================#

APP_TITLE=LibreChat
# CUSTOM_FOOTER=""

DEBUG_LOGGING=true

HOST=192.168.1.xxx
PORT=3080

MONGO_URI=mongodb://127.0.0.1:27017/LibreChat

DOMAIN_CLIENT=http://192.168.1.xxx:3080
DOMAIN_SERVER=http://192.168.1.xxx:3080

#=============#
# Permissions #
#=============#

UID=10xx
GID=10xx

#============#
# OpenAI     #
#============#

OPENAI_API_KEY=xxxxxxxxxx
OPENAI_MODELS=gpt-3.5-turbo-1106,gpt-4-1106-preview,gpt-3.5-turbo,gpt-3.5-turbo-16k,gpt-3.5-turbo-0301,text-davinci-003,gpt-4,gpt-4-0314,gpt-4-0613,gpt-4-vision-preview,

DEBUG_OPENAI=false

# TITLE_CONVO=false
# OPENAI_TITLE_MODEL=gpt-3.5-turbo

# OPENAI_SUMMARIZE=true
# OPENAI_SUMMARY_MODEL=gpt-3.5-turbo

# OPENAI_FORCE_PROMPT=true

# OPENAI_REVERSE_PROXY=

# DALL·E 3
#----------------
DALLE_API_KEY=xxxxxxxx
# DALLE3_SYSTEM_PROMPT="Your System Prompt here"
# DALLE_REVERSE_PROXY=
```

**Comment by fuegovic** [comment]
It's working fine for me, if you are not aware of reverse proxies you are probably not using one, so no issue there. The most confusing thing to me in your screenshot is the fact that GPT-3.5 has the Google PaLM logo, this isn't the normal behavior 🤔

![image](https://github.com/danny-avila/LibreChat/assets/32828263/8f9201ea-116e-44ac-8126-486c1d788e76)

**Reply by points-unknown** [reply]
Thanks for the quick response. Yeah - it seems to be hit and miss. Is there a rate limit? Like I can only do one image every 10 minutes?

Because, I was inspired by your image so I did this:
![image](https://github.com/danny-avila/LibreChat/assets/34283742/8a080f39-c381-4f11-93d3-8d674374f78c)

And then I tried my original prompt again:

![image](https://github.com/danny-avila/LibreChat/assets/34283742/45e79289-863a-4fb7-9213-f8f16e9616ab)

**Reply by fuegovic** [reply]
Are you on a paid or free plan for the OpenAI API? The free plan has much lower limits than the paid ones, which could explain your results.

(Cool pic btw)

**Reply by points-unknown** [reply, marked as answer]
The issue appears to be with the word "4K" in the request. The only thing I can think of is that it is interpreting this as a size request - which is obviously too big.

![image](https://github.com/danny-avila/LibreChat/assets/34283742/411ecb17-b199-4301-8137-a6c67f3b3c24)

![image](https://github.com/danny-avila/LibreChat/assets/34283742/d70ad362-5f33-43ea-9a4e-7424b1ee3985)

**Reply by fuegovic** [reply]
I think you're right

![image](https://github.com/danny-avila/LibreChat/assets/32828263/549747ba-3d35-45ff-864a-95ad63770d41)

**Comment by fuegovic** [comment]
Converted back to issues -> #1410

@points-unknown thanks for investigating this, I will close the discussion in favor of the issue
